codegen/module.go

```diff
@@ -284,18 +298,6 @@ func (system *ModuleSystem) ResolveModules(
 		if err != nil {
 			return nil, err
 		}
-
-		// Resolve dependencies again in case two instances
```

Comment:
No that's not what resolving is. Resolving means taking the Dependencies and turning them into ResolvedDependencies, by pulling out the module instance from the global list of module instance. So Dependencies in a map of Class to InstanceName, something like

client: ["foo", "bar", "baz"]

The ResolveDependencies function is building in reverse DAG order the dependencies, so it reads the clients, resolves the client dependencies, reads the business dir, resovles the dependencies, reads the endpoints, resolves the endpoint dependencies.

Because we do it in order, you should only have to read once and resolve once. It's just replacing the instance name with the instance object that we read. The reason why we can't read and resolve at the same time is because if client Foo depends on client Bar, it might not exist yet. A missing dependency in this resolve function should be an error, because there is no chance of it being read in future. Once we had read all clients, all clients will be resolvable.